parfive/downloader.py

```diff
@@ -172,11 +176,21 @@ def enqueue_file(
             and the existing file will be overwritten, if `'unique'` the filename
             will be modified to be unique. If `None` the value set when
             constructing the `~parfive.Downloader` object will be used.
+        checksum
+            If given the downloaded file will be verified against the
+            given checksum.  The format of the checksum string should
+            be ``<algorithm>=<checksum>``, valid algorithms are
+            ``sha-512``, ``sha-256``, ``md5`` or ``sha``.  If ``True``
+            then where provided by the server (in the ``Repr-Digest``
+            or ``Content-Digest`` headers) the checksum will be
+            validated against the checksum returned by the server.
```

Comment:
I think it'd be worth clarifying some of the behavior we discussed relating to pre-existing local files. For instance:

* If a local file already exists and checksum is set to an explicit value, then the download will be skipped if the checksum of the local file is correct
* If a local file already exists and checksum is set to ``True``, then the download will be skipped if the server reports a checksum that is the same as an existing local file
* If no local files exist and the server reports a checksum that is not the one the user requested, the download happens *anyway* and then will likely fail once the file is actually downloaded if the hash is not correct.

The last one should probably be changed because it seems weird to sometimes trust the server hash and sometimes not.

Reply:
I have now changed the last one, and I think clarified the docstring for the other two.